webapp/components/test-file-results-time-series.html

```diff
@@ -0,0 +1,135 @@
+      }
+      .browser-result-status.FAIL {
+        background-color: rgb(242, 90, 90);
+      }
+    </style>
+    <template is="dom-repeat" items="[[results]]" as="browserResults">
+      <div class="browser">
+        <div class="browser-name">[[browserResults.browserName]]</div>
+        <div class="browser-results"></div>
+        <template is="dom-repeat" items="[[browserResults.runResults]]" as="runResult">
+            <div class="browser-result">
+              <!-- Tooltip or something associated with run? -->
+              <template is="dom-repeat" items="[[runResult.results]]" as="result">
+                <div
+                    class$="browser-result-status [[result.status]]"
+                    onmouseenter="[[bindResultHover(runResult.run, result.name)]]">&nbsp;</div>
+              </template>
+            </div>
+        </template>
+      </div>
+    </template>
+
+    <pre id="status">&nbsp;</pre>
+  </template>
+  <script>
+    /* global TestRunsBase */
+    class TestFileResultsTimeSeries extends TestRunsBase {
+      static get is() {
+        return 'test-file-results-time-series';
+      }
+
+      static get properties() {
+        return {
+          maxCount: {
+            type: Number,
+            value: 20,
+          },
+          labels: {
+            type: Array,
+            value: [],
+          },
+          results: {
+            type: Array,
+            value: [],
+          },
+        };
+      }
+
+      static get observers() {
+        return [
+          // Observe testRuns bound to component. Expect each to have its
+          // subtests property eventually filled out (as in TestFileResults)
+          // logic).
+          'onTestRuns(testRuns)',
+        ];
+      }
+
+      // on-status-box-hover: Update status tag with summary of source run+test.
+      bindResultHover(run, subTestName) {
+        return () => {
+          let statusElement = this.shadowRoot.querySelector('#status');
+          statusElement.textContent =
+              `${run.browser_name} ${run.browser_version} ${run.os_name} ${run.os_version} @ ${run.time_start} : ${subTestName}`;
+
+        };
+      }
+
+      onTestRuns(testRuns) {
+        // Reduce this.testRuns, which may have testRun.subtests filled out
+        // (or may not) to browserName grouped structure:
+        // [{browserName, runResults: [{run, results: [resultStatus]}]}].
+        this.results = testRuns.reduce((acc, run) => {
+          const browserResultsIdx = acc
+            .findIndex(br => br.browserName === run.browser_name);
+          let browserResults = acc[browserResultsIdx];
+          if (!browserResults) {
+            browserResults = {
+              browserName: run.browser_name,
+              runResults: [],
+            };
+            acc.push(browserResults);
+          }
+          browserResults.runResults.push({
+            run,
+            results: run.subtests ? Object.keys(run.subtests)
```

Comment:
Nit: Wrap ternary like this

    run.subtests
        ? foo({...
            ...
          })
        : ...